build/olm-catalog/yaml-options-gen.py

```diff
@@ -0,0 +1,432 @@
+        elif option['type'] == 'Dict of String':
+            self.form_type = None
+            self.ignore = True
+            self.default = option['default']
+            sys.stderr.write('Ignoring %s with type %s\n' % (option['name'],
+                                                             option['type']))
+        elif option['type'] == 'List of String':
+            self.form_type = None
+            self.ignore = True
+            self.default = option['default']
+            sys.stderr.write('Ignoring %s with type %s\n' % (option['name'],
+                                                             option['type']))
+        elif option['type'] == 'List of IPAddress':
+            self.form_type = None
+            self.ignore = True
+            self.default = option['default']
+            sys.stderr.write('Ignoring %s with type %s\n' % (option['name'],
+                                                             option['type']))
+
+        else:
+            raise Exception('Unkown type %s' % option['type'])
+
+        if option['required'] == 'True':
+            self.help += ' [Optional]'
+
+    def add_driver(self, driver):
+        self.drivers.add(driver)
+
+
+def include_driver(drivername):
+    if EXCLUDE_DRIVERS is not None:
+        for regex in EXCLUDE_DRIVERS:
+            if re.match(regex, drivername, re.IGNORECASE):
+                return False
+
+    if INCLUDE_DRIVERS is None:
+        return True
+
+    for regex in INCLUDE_DRIVERS:
+        if re.match(regex, drivername, re.IGNORECASE):
+            return True
+
+    return False
+
+
+def render_option(option, driver, group_options=False):
+    if option.ignore:
+        return ''
+    group = driver + "Options" if group_options else 'DriverSettings'
+
+    result = """
+        - description: {description}
+          displayName: {display_name}
+          path: config.envVars.X_CSI_BACKEND_CONFIG.{name}
+          x-descriptors:
+            - 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:{group}'
+            - {opt_type}
+            - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:config.envVars.X_CSI_BACKEND_CONFIG.driver:{driver}'
+    """.format(name=option.name,  # noqa
+               display_name=option.display_name,
+               description='"' + option.help + '"',
+               opt_type=option.form_type,
+               driver=driver,
+               group=group)
+
+    # The only way to present the groups folded is to set them as advanced
+    if group_options:
+        result += 12 * ' ' + "- 'urn:alm:descriptor:com.tectonic.ui:advanced'"
+    # We don't indent here, it's the caller's responsibility
+    return result
+
+
+def additional_options():
+    """Options that are generic for all drivers."""
+
+    # TODO: Make the operator automatically set X_CSI_BACKEND_CONFIG.name and
+    # X_CSI_EMBER_CONFIG.plugin_name based on the deployment name.
+    res = """        - description: Backend name
```

Comment:
I think this should start with a newline

Reply:
You can see in the output that it's not necessary: https://github.com/embercsi/ember-csi-operator/blob/6029712b36a0454f4d464a3c5b000c0b9cddbdd3/deploy/olm-catalog/next/ember-csi-operator.vX.Y.Z.clusterserviceversion.yaml#L3041

That's because each rendered option finishes with a new line: https://github.com/embercsi/ember-csi-operator/blob/6029712b36a0454f4d464a3c5b000c0b9cddbdd3/build/olm-catalog/yaml-options-gen.py#L244

Reply:
Thanks for pointing out that I was looking at the wrong place:
https://github.com/embercsi/ember-csi-operator/blob/6029712b36a0454f4d464a3c5b000c0b9cddbdd3/deploy/olm-catalog/next/ember-csi-operator.vX.Y.Z.clusterserviceversion.yaml#L7488